src/main/java/com/avispl/symphony/dal/infrastructure/management/tripleplay/TriplePlayAggregatorCommunicator.java

```diff
@@ -132,8 +132,11 @@ private void retrieveClient(List<String> clientIps) {
 						aggregatedDevice.setDeviceId(getDefaultValueForNullOrEmpty(client.getClientId()));
 						aggregatedDevice.setProperties(properties);
 						aggregatedDevice.setDeviceName(getDefaultValueForNullOrEmpty(client.getTypeDescription()));
-						aggregatedDevice.setDeviceOnline(client.getConnectionStatus().equals(TriplePlayConstrant.ONLINE));
-						aggregatedDevice.setType(getDefaultValueForNullOrEmpty(client.getType()));
+						if (client.getConnectionStatus() != null) {
+							aggregatedDevice.setDeviceOnline(client.getConnectionStatus().equals(TriplePlayConstrant.ONLINE));
```

Comment:
comparing with == would be null safe in this case :) will "getConnectionStatus" always be non-null?